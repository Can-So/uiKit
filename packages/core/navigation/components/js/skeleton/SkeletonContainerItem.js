import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import React, { Component } from 'react';
import { Skeleton as SkeletonIcon } from '@findable/icon';
import { HiddenWhenCollapsed } from './ToggleWhenCollapsed';
import SkeletonContainerItemWrapper from './styled/SkeletonContainerItemWrapper';
import SkeletonContainerItemText from './styled/SkeletonContainerItemText';
import SkeletonIconWrapper from './styled/SkeletonIconWrapper';

var SkeletonContainerItem =
/*#__PURE__*/
function (_Component) {
  _inherits(SkeletonContainerItem, _Component);

  function SkeletonContainerItem() {
    _classCallCheck(this, SkeletonContainerItem);

    return _possibleConstructorReturn(this, _getPrototypeOf(SkeletonContainerItem).apply(this, arguments));
  }

  _createClass(SkeletonContainerItem, [{
    key: "render",
    value: function render() {
      return React.createElement(SkeletonContainerItemWrapper, null, React.createElement(SkeletonIconWrapper, null, React.createElement(SkeletonIcon, null)), React.createElement(HiddenWhenCollapsed, {
        isCollapsed: this.props.isCollapsed
      }, React.createElement(SkeletonContainerItemText, {
        textWidth: this.props.itemTextWidth
      })));
    }
  }]);

  return SkeletonContainerItem;
}(Component);

_defineProperty(SkeletonContainerItem, "defaultProps", {
  isCollapsed: false
});

export { SkeletonContainerItem as default };