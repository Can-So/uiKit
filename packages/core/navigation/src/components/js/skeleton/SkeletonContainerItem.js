// @flow
import React, { Component } from 'react';

import { Skeleton as SkeletonIcon } from '@atlaskit/icon';

import { HiddenWhenCollapsed } from './ToggleWhenCollapsed';

import SkeletonContainerItemWrapper from './styled/SkeletonContainerItemWrapper';
import SkeletonContainerItemText from './styled/SkeletonContainerItemText';
import SkeletonIconWrapper from './styled/SkeletonIconWrapper';

type Props = {
  isCollapsed?: boolean,
  itemTextWidth?: string,
};

export default class SkeletonContainerItem extends Component<Props> {
  static defaultProps = {
    isCollapsed: false,
  };

  render() {
    return (
      <SkeletonContainerItemWrapper>
        <SkeletonIconWrapper>
          <SkeletonIcon />
        </SkeletonIconWrapper>
        <HiddenWhenCollapsed isCollapsed={this.props.isCollapsed}>
          <SkeletonContainerItemText textWidth={this.props.itemTextWidth} />
        </HiddenWhenCollapsed>
      </SkeletonContainerItemWrapper>
    );
  }
}
