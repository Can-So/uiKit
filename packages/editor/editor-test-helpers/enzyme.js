import * as tslib_1 from "tslib";
/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */
import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
// Create the IntlProvider to retrieve context for wrapping around.
var intlProvider = new IntlProvider({ locale: 'en' });
var intl = intlProvider.getChildContext().intl;
/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
function nodeWithIntlProp(node) {
    return React.cloneElement(node, { intl: intl });
}
export function shallowWithIntl(node, _a) {
    if (_a === void 0) { _a = {}; }
    var _b = _a.context, context = _b === void 0 ? {} : _b, additionalOptions = tslib_1.__rest(_a, ["context"]);
    if (typeof node.type !== 'string' && node.type.name === 'InjectIntl') {
        var unwrappedType = node.type.WrappedComponent;
        node = React.createElement(unwrappedType, node.props);
    }
    return shallow(nodeWithIntlProp(node), tslib_1.__assign({ context: tslib_1.__assign({}, context, { intl: intl }) }, additionalOptions));
}
export function mountWithIntl(node, _a) {
    if (_a === void 0) { _a = {}; }
    var _b = _a.context, context = _b === void 0 ? {} : _b, _c = _a.childContextTypes, childContextTypes = _c === void 0 ? {} : _c, additionalOptions = tslib_1.__rest(_a, ["context", "childContextTypes"]);
    if (typeof node.type !== 'string' && node.type.name === 'InjectIntl') {
        var unwrappedType = node.type.WrappedComponent;
        node = React.createElement(unwrappedType, node.props);
    }
    return mount(nodeWithIntlProp(node), tslib_1.__assign({ context: tslib_1.__assign({}, context, { intl: intl }), childContextTypes: tslib_1.__assign({}, childContextTypes, { intl: intlShape }) }, additionalOptions));
}
//# sourceMappingURL=enzyme.js.map