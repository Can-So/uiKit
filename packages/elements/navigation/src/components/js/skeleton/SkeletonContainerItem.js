// @flow
import React, { Component } from 'react';

import { Icon, Paragraph } from '@atlaskit/skeleton';

import { HiddenWhenCollapsed } from './ToggleWhenCollapsed';

import SkeletonContainerItemWrapper from '../../styled/skeleton/SkeletonContainerItemWrapper';
import SkeletonContainerItemText from '../../styled/skeleton/SkeletonContainerItemText';

type Props = {
  isCollapsed: boolean,
};

export default class SkeletonContainerItem extends Component<Props> {
  static defaultProps = {
    isCollapsed: false,
  };
  render() {
    return (
      <SkeletonContainerItemWrapper>
        <Icon />
        <HiddenWhenCollapsed isCollapsed={this.props.isCollapsed}>
          <SkeletonContainerItemText>
            <Paragraph />
          </SkeletonContainerItemText>
        </HiddenWhenCollapsed>
      </SkeletonContainerItemWrapper>
    );
  }
}
