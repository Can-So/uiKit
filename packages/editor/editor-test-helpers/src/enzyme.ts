/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */
import * as React from 'react';
import { ReactElement } from 'react';
import { IntlProvider, InjectedIntlProps, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme';

// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider({ locale: 'en' });
const { intl } = intlProvider.getChildContext();

/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl });
}

export function shallowWithIntl<P>(
  node: ReactElement<P>,
  { context = {}, ...additionalOptions } = {},
) {
  if (typeof node.type !== 'string' && node.type.name === 'InjectIntl') {
    const unwrappedType = (node.type as any).WrappedComponent;
    (node as any) = React.createElement(unwrappedType, node.props);
  }
  return shallow(
    nodeWithIntlProp(node) as ReactElement<P & InjectedIntlProps>,
    {
      context: { ...context, intl },
      ...additionalOptions,
    },
  );
}

export function mountWithIntl<P>(
  node: ReactElement<P>,
  { context = {}, childContextTypes = {}, ...additionalOptions } = {},
) {
  if (typeof node.type !== 'string' && node.type.name === 'InjectIntl') {
    const unwrappedType = (node.type as any).WrappedComponent;
    (node as any) = React.createElement(unwrappedType, node.props);
  }
  return mount(nodeWithIntlProp(node) as ReactElement<P & InjectedIntlProps>, {
    context: { ...context, intl },
    childContextTypes: { ...childContextTypes, intl: intlShape },
    ...additionalOptions,
  });
}
