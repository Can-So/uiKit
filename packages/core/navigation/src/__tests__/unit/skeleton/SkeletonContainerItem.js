// @flow
import React from 'react';
import { shallow } from 'enzyme';

import SkeletonContainerItem from '../../../components/js/skeleton/SkeletonContainerItem';

test('renders a skeleton representation of a container item', () => {
  expect(shallow(<SkeletonContainerItem />)).toMatchSnapshot();
});

test('shows a compact version when collapsed', () => {
  expect(shallow(<SkeletonContainerItem isCollapsed />)).toMatchSnapshot();
});

test('passes down custom styles', () => {
  expect(
    shallow(<SkeletonContainerItem itemTextWidth="100%" />),
  ).toMatchSnapshot();
});
