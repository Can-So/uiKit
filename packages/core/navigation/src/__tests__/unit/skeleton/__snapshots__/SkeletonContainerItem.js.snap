// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes down custom styles 1`] = `
<SkeletonContainerItemWrapper>
  <SkeletonIconWrapper>
    <Skeleton
      size="medium"
      weight="normal"
    />
  </SkeletonIconWrapper>
  <HiddenWhenCollapsed
    isCollapsed={false}
  >
    <SkeletonContainerItemText
      textWidth="100%"
    />
  </HiddenWhenCollapsed>
</SkeletonContainerItemWrapper>
`;

exports[`renders a skeleton representation of a container item 1`] = `
<SkeletonContainerItemWrapper>
  <SkeletonIconWrapper>
    <Skeleton
      size="medium"
      weight="normal"
    />
  </SkeletonIconWrapper>
  <HiddenWhenCollapsed
    isCollapsed={false}
  >
    <SkeletonContainerItemText />
  </HiddenWhenCollapsed>
</SkeletonContainerItemWrapper>
`;

exports[`shows a compact version when collapsed 1`] = `
<SkeletonContainerItemWrapper>
  <SkeletonIconWrapper>
    <Skeleton
      size="medium"
      weight="normal"
    />
  </SkeletonIconWrapper>
  <HiddenWhenCollapsed
    isCollapsed={true}
  >
    <SkeletonContainerItemText />
  </HiddenWhenCollapsed>
</SkeletonContainerItemWrapper>
`;
